resolwe/flow/executors/startup_communication_container.py

```diff
@@ -0,0 +1,392 @@
+
+    async def start_processing_socket(self):
+        """Start listening on the processing file socket.
+
+        The socket is used by the processing container to communicatite with
+        us.
+        """
+        await asyncio.start_unix_server(
+            self._handle_processing_container_connection, os.fspath(PROCESSING_SOCKET)
+        )
+        logger.debug("Started listening on %s.", PROCESSING_SOCKET)
+
+    async def open_listener_connection(self) -> ZMQCommunicator:
+        """Connect to the listener service.
+
+        We are using data id as identity. This implies only one process per
+        data object at any given point in time can be running.
+        """
+        zmq_context = zmq.asyncio.Context.instance()
+        zmq_socket = zmq_context.socket(zmq.DEALER)
+        zmq_socket.setsockopt(zmq.IDENTITY, str(DATA_ID).encode())
+        connect_string = f"{LISTENER_PROTOCOL}://{LISTENER_IP}:{LISTENER_PORT}"
+        logger.debug("Opening listener connection to %s", connect_string)
+        zmq_socket.connect(connect_string)
+        return ZMQCommunicator(zmq_socket, "worker <-> listener", logger)
+
+    async def transfer_missing_data(self):
+        """Transfer missing data.
+
+        Log error re-raise exception on failure.
+
+        :raises: RuntimeError on failure.
+        """
+        try:
+            await transfer_data(self.listener_communicator)
+        except RuntimeError:
+            with suppress(Exception):
+                await self.listener_communicator.send_command(
+                    Message.command(
+                        "process_log", {"error": ["Error transfering missing data."]}
+                    )
+                )
+            raise
+
+    async def collect_produced_files(self):
+        """Collect files produced by the worker.
+
+        Keep only files that are referenced in the data model.
+        Log error re-raise exception on failure.
+
+        :raises: RuntimeError on failure.
+        """
+        try:
+            logger.debug("Collecting files")
+            await collect_files(self.listener_communicator)
+            logger.debug("Collected files")
+            return True
+        except RuntimeError:
+            with suppress(Exception):
+                await self.listener_communicator.send_command(
+                    Message.command(
+                        "process_log",
+                        {"error": ["Error collecting produced files."]},
+                    )
+                )
+        return False
+
+    def _communicator_stopped(self, future: asyncio.Future):
+        """Stop processing if necessary."""
+        if self._process_script_task:
+            logger.debug("Communicator closed, cancelling script processing.")
+            self._process_script_task.cancel()
+
+    async def start(self) -> int:
+        """Start the main program."""
+        try:
+            return_code = 1
+            await self.start_processing_socket()
+<<<<<<< HEAD
```

Comment:
Easy: communication container is used as image in testing.
I will fix tests so that the com container will be build first and then tests are run using it.